src/hooks/useNextEventWidget.ts

```
@@ -0,0 +1,188 @@
+
+export function useNextEventWidget(): UseNextEventWidgetReturn {
+    const [event, setEvent] = useState<NextEventWidgetData | null>(null);
+    const [countdown, setCountdown] = useState<CountdownState | null>(null);
+    const [isLoading, setIsLoading] = useState(true);
+    const [error, setError] = useState<string | null>(null);
+    const [isEmpty, setIsEmpty] = useState(false);
+    const [emptyMessage, setEmptyMessage] = useState(EMPTY_STATE_MESSAGE);
+
+    const timerRef = useRef<NodeJS.Timeout | null>(null);
+
+    const calculateCountdown = useCallback((startDate: string, endDate: string): CountdownState => {
```

ESLint

[error] Replace `····` with `··`

(prettier/prettier)